src/validation.cpp

```diff
@@ -3464,9 +3464,7 @@ static FILE* OpenDiskFile(const CDiskBlockPos &pos, const char *prefix, bool fRe
         return nullptr;
     fs::path path = GetBlockPosFilename(pos, prefix);
     fs::create_directories(path.parent_path());
-    FILE* file = fsbridge::fopen(path, "rb+");
-    if (!file && !fReadOnly)
-        file = fsbridge::fopen(path, "wb+");
+    FILE* file = fsbridge::fopen(path, fReadOnly ? "rb": "ab+");
```

Comment:
Don't some operating systems enforce the append bit, so that you can really only append?
From the man page:
```
       a+     Open for reading and appending (writing at end of file).  The file is created if it does not exist.  The initial  file  position  for
              reading is at the beginning of the file, but output is always appended to the end of the file.
```
I don't think that's quite what we want.